blocks/juxtapose/src/edit.js

```diff
@@ -28,53 +29,57 @@ const edit= ({ attributes, setAttributes, className }) => {
 						setAttributes( { orientation: val } );
 						// need slight delay so markup can be updated before
 						// scan page gets triggered
-						setTimeout( function() { juxtapose.scanPage(); }, 100 );
+						setTimeout( function() {
+							juxtapose.scanPage();
+						}, 100 );
 					} }
 				/>
 			</InspectorControls>
-			<div className={cls} data-mode={orientation}>
+			<div className={ cls } data-mode={ orientation }>
 
-				{ imageBefore ? (
-					<img src={imageBefore} />
-				) : (
-					<div className='img-edit-before'>
-						<MediaPlaceholder
-							onSelect = {
-								( el ) => {
-									setAttributes( { imageBefore: el.url } );
-									juxtapose.scanPage();
+				<Placeholder>
+					{ imageBefore ? (
+						<img src={ imageBefore } />
+					) : (
+						<div className="img-edit-before">
+							<MediaPlaceholder
+								onSelect ={
+									( el ) => {
+										setAttributes( { imageBefore: el.url } );
+										juxtapose.scanPage();
+									}
 								}
-							}
-							allowedTypes = {[ 'image' ]}
-							labels ={ {title: 'Image Before'} }
-						/>
-					</div>
-				)}
+								allowedTypes={ [ 'image' ] }
+								labels={ { title: 'Image Before' } }
+							/>
+						</div>
+					) }
 
-				{ imageAfter ? (
-					<img src={imageAfter} />
-				) : (
-					<div className='img-edit-after'>
-						<MediaPlaceholder
-							onSelect = {
-								( el ) => {
-									setAttributes( { imageAfter: el.url } );
-									juxtapose.scanPage();
+					{ imageAfter ? (
+						<img src={ imageAfter } />
+					) : (
+						<div className="img-edit-after">
+							<MediaPlaceholder
+								onSelect={
+									( el ) => {
+										setAttributes( { imageAfter: el.url } );
+										juxtapose.scanPage();
+									}
 								}
-							}
-							allowedTypes = {[ 'image' ]}
-							labels = { {title: 'Image After'} }
-						/>
-					</div>
-				)}
-			</div>
-			<div className='caption'>
-				<TextControl
-					placeholder="Write caption…"
-					onChange={ (val) => setAttributes({ caption: val }) }
-					value={caption}
-				/>
+								allowedTypes={ [ 'image' ] }
+								labels={ { title: 'Image After' } }
+							/>
+						</div>
+					) }
+				</Placeholder>
 			</div>
+			<TextControl
```

Comment:
I tried making this a `RichText` component, that was only visible when the block was `isSelected`, but failed. Feel free to take a stab. As you can see I added an `inlineToolbar` prop for bold, etc.

Reply:
I pushed an update to switch caption to RichText 👍